### PR TITLE
Internal Standardization & Automation Improvements

### PR DESCRIPTION
**Pull request by agilgur5**
99% of the code in this repo is ~3.5 years old, so this PR starts upgrading everything to be more consistent with current styles and the way I do most of my other libraries (that have been updated much more recently).

- `.js` instead of `.es6`
- `files` instead of `.npmignore`
- better organized `package.json` fields and more fields filled
- better organized and named `package.json` scripts
- standardize badges

And some automation improvements in the same vein:
- `CHANGELOG.md` and `changelog` script
  - I also updated the [releases page](https://github.com/agilgur5/trim-canvas/releases) with mostly auto-generated changelogs. Changed a bit of what's in the "Release" sections as I didn't use `pub` commits way back when so the auto-generated ones didn't make as much sense as a result
- CI support and `test:pub` script to test build + pack
- `pub` script

Basically lots of quality improvements other than/prior to adding tests and changing the ~3.5 year old build process -- both of which are coming very soon!

### CODE REVIEW

**Review by agilgur5 — COMMENTED** [review]
Fairly small diff, looks all good to me!
